src/core/app.lua

```diff
@@ -51,10 +51,13 @@ local function with_restart (app, method)
 end
 
 -- Restart dead apps.
+local actions = { start={}, restart={}, reconfig={}, keep={}, stop={} }
 function restart_dead_apps ()
+   if not use_restart then return end
    local restart_delay = 2 -- seconds
-   local actions = { start={}, restart={}, reconfig={}, keep={}, stop={} }
    local restart = false
+   actions.restart[1] = nil
```

Comment:
Oops, that was a little simple-minded by me. It will also not be reflected in #actions.restart. I guess the proper way to clear the array is to traverse it and set all elements to nil.

I'm sure this is a Lua FAQ and I fell right into the trap :)